### PR TITLE
Improve codebase

### PR DESCRIPTION
**Pull request by eggplants**
Resolves: #12

## Summary

- introduced `poetry` and `pre-commit`
- replaced `setup.py` with `pyproject.toml`
- replaced Travis CI with GitHub Actions
  - [repo secrets](https://docs.github.com/en/actions/security-for-github-actions/security-guides/using-secrets-in-github-actions#creating-secrets-for-a-repository) (`TESTPYPI_API_TOKEN`, `PYPI_API_TOKEN`) needed
- formatted with black and ruff
- fix (and suppress) ruff and markdownlint-cli errors

## How to develop

```bash
# python>=3.9
pip install pipx
pipx install poetry
poetry install
poetry shell
pre-commit install

# run
# if not set any PyPI token `poetry publish` fails
python publish.py
```

```shellsession
$ openapi-generator-cli --help
usage: openapi-generator-cli <command> [<args>]

The most commonly used openapi-generator-cli commands are:
    author        Utilities for authoring generators or customizing templates.
    batch         Generate code in batch via external configs.
    config-help   Config help for chosen lang
    generate      Generate code with the specified generator.
    help          Display help information about openapi-generator
    list          Lists the available generators
    meta          MetaGenerator. Generator for creating a new template set and configuration for Codegen.  The output will be based on the language you specify, and includes default templates to include.
    validate      Validate specification
    version       Show version information used in tooling

See 'openapi-generator-cli help <command>' for more information on a specific
command.
$ openapi-generator-cli version
7.10.0
$ java --version
openjdk 23.0.1 2024-10-15
OpenJDK Runtime Environment Homebrew (build 23.0.1)
OpenJDK 64-Bit Server VM Homebrew (build 23.0.1, mixed mode, sharing)
```

## Notes

Currently, the command name is `openapi-generator`, but it should be available in `openapi-generator-cli` as well as in [the node wrapper](https://github.com/OpenAPITools/openapi-generator-cli?tab=readme-ov-file#usage).

After merging this PR, I'm going to add some unit tests and its testing workflow (#12).

### CODE REVIEW

**Comment by wing328** [comment]
looks good to me, we can give it a try

a little background on https://github.com/openAPITools/openapi-generator-cli/ ... original the command was named `openapi-generator` in CLI 1.x but later renamed to `openapi-generator-cli` in CLI 2.x release. one goal is to avoid the confusion with the `openapi-generator` command installed via `brew install openapi-generator`

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+python = ">=3.9,<4"
+
+[tool.poetry.group.dev.dependencies]
+mypy = ">=0.991,<1.14"
+pre-commit = ">=2.20,<4.0"
+taskipy = "^1.10.3"
+pytest = ">=7.2.2,<9.0.0"
+
+[tool.poetry.scripts]
+openapi-generator = "openapi_generator_cli:cli"
```
Comment (eggplants) [inline pyproject.toml:49]: @wing328 Could I change this into `openapi-generator-cli`?
```suggestion
[tool.poetry.scripts]
openapi-generator-cli = "openapi_generator_cli:cli"
```

File: .github/workflows/publish.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+name: Release Package
+
+# need to setup:
+# secrets.PYPI_API_TOKEN (See: https://pypi.org/help/#apitoken)
+# secrets.TESTPYPI_API_TOKEN (See: https://test.pypi.org/help/#apitoken)
```
Comment (eggplants) [inline .github/workflows/publish.yaml:5]: > [repo secrets](https://docs.github.com/en/actions/security-for-github-actions/security-guides/using-secrets-in-github-actions#creating-secrets-for-a-repository) (`TESTPYPI_API_TOKEN`, `PYPI_API_TOKEN`) needed

I do not have the authority to set this up.
@wing328 If I send these to DM via Slack, can you set it up?

Comment (wing328) [reply .github/workflows/publish.yaml:5]: sure

File: pyproject.toml (hunk 1 above)
Comment (wing328) [reply pyproject.toml:49]: up to you

File: .github/workflows/publish.yaml (hunk 2 above)
Comment (eggplants) [reply .github/workflows/publish.yaml:5]: Sent. Please check!

File: pyproject.toml (hunk 1 above)
Comment (eggplants) [reply pyproject.toml]: Changed in 3a7b83b

File: .github/workflows/publish.yaml (hunk 2 above)
Comment (wing328) [reply .github/workflows/publish.yaml:5]: done

Comment (eggplants) [reply .github/workflows/publish.yaml:5]: To get environment secrets its name are required. So please tell me it.
usage:　https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#example-using-a-single-environment-name

**Pull request merged**